src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -239,32 +243,34 @@ private Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibility)
                             var name = rule.NamingStyle.CreateName(baseName).EscapeIdentifier(context.IsInQuery);
 
                             // Don't add multiple items for the same name and only add valid identifiers
-                            if (name.Length > 1 && CSharpSyntaxFacts.Instance.IsValidIdentifier(name) && !result.ContainsKey(name))
+                            if (name.Length > 1 &&
+                                CSharpSyntaxFacts.Instance.IsValidIdentifier(name) &&
+                                seenBaseNames.Add(name))
                             {
-                                var targetToken = context.TargetToken;
                                 var uniqueName = semanticFactsService.GenerateUniqueName(
                                     context.SemanticModel,
                                     context.TargetToken.Parent,
                                     containerOpt: null,
                                     baseName: name,
-                                    filter: IsRelevantSymbolKind,
+                                    filter: s => IsRelevantSymbolKind(s),
```

Comment:
Why explicitly create a lambda?

Reply:
because lambdas can be cached, method groups cannot be.  so this saves on unnecessary allocations.